block/fork.go

```diff
@@ -250,41 +249,57 @@ func (m *Manager) updateStateWhenFork() error {
 	return nil
 }
 
-// forkFromInstruction checks if fork is needed, reading instruction file, and performs fork actions
-func (m *Manager) forkFromInstruction() error {
-	// if instruction file exists proposer needs to do fork actions (if settlement height is higher than revision height it is considered fork already happened and no need to do anything)
-	instruction, forkNeeded := m.forkNeeded()
-	if !forkNeeded {
-		return nil
+// checkRevisionAndFork checks if fork is needed after syncing, and performs fork actions
+func (m *Manager) checkRevisionAndFork() error {
+	// it is checked again whether the node is the active proposer, since this could have changed after syncing.
+	amIProposerOnSL, err := m.AmIProposerOnSL()
+	if err != nil {
+		return fmt.Errorf("am i proposer on SL: %w", err)
+	}
+	if !amIProposerOnSL {
+		return fmt.Errorf("the node is no longer the proposer. please restart.")
+	}
+
+	// update sequencer in case it changed after syncing
+	err = m.UpdateProposerFromSL()
+	if err != nil {
+		return err
+	}
```

Comment:
the first one is just a check that nothing changed (the node is still the proposer) between when the node started  and finished syncing (that could be days). the second is updating dymint state proposer with the current proposer from SL. in case it exits before, it does not really matter because it will start as a fullnode and it will update it anyway on start if its at the fork height.